_drafts/code-properties-and-language-features.md

```diff
@@ -0,0 +1,50 @@
+I feel like I only just started learning good object-oriented design, and the world has moved on to functional programming! As a result, the OO principles that are fresh on my mind are exactly what people are moving away from. For example, as I've been checking out Elixir and Redux, I've had questions about whether a lack of certain OO features causes problems. For example, encapsulation allows you to isolate implementation knowledge to one part of your app, so that other parts of your app can use that functionality without being coupled to its implementation. Is it a problem that FP doesn't offer the same kind of encapsulation?
```

Comment:
I think I wasn't sufficiently precise about the meaning of "encapsulation" I was using here. There are multiple uses: in fact, the Wikipedia article on "encapsulation" has two definitions, neither of which is what I meant here!

The meaning I intended matches what's in [the Wikipedia article on OOP](https://en.wikipedia.org/wiki/Object-oriented_programming#Encapsulation) (emphasis mine):

> Encapsulation is an object-oriented programming concept that binds together the data and functions that manipulate the data, and that keeps both safe from outside interference and misuse. Data encapsulation led to the important OOP concept of data hiding. If a class _does not allow calling code to access internal object data_ and permits access through methods only, this is a strong form of abstraction or information hiding known as encapsulation.

Clarified form of this sentences I'm considering:

> For example, encapsulation allows you to protect an object's internal state from being accessed from elsewhere in the app, so that you can change the implementation of the data storage with the confidence that the rest of the app isn't affected.

Does that statement seem more correct? (Again, part of the point later in the article is that you can get properties _like_ these in Elixir by convention.)